test/mbed_gt_cmake_handlers.py

```diff
@@ -16,8 +16,17 @@
 limitations under the License.
 """
 
+import os
+import mock
+import sys
 import unittest
-from mbed_greentea import cmake_handlers
+
+try:
+    from StringIO import StringIO
```

Comment:
`StringIO` is also available in Python 2's `io` package, so I think you can do away with this try/except block

https://docs.python.org/2/library/io.html#io.StringIO

Reply:
Ah, you're right, I'll change that. I forgot to test with python 3 before making the PR so I ended up going through adding fixes to things that were breaking the CI.

Reply:
No worries, I just learned about Python 2's `io` package a couple days ago myself!

Reply:
Annoyingly, the types they use in each version don't seem to compatible for this case

```
Python 3.6.2 (v3.6.2:5fd33b5, Jul  8 2017, 04:57:36) [MSC v.1900 64 bit (AMD64)] on win32
>>> from io import StringIO
>>> print(type(StringIO().getvalue()))
<class 'str'>

Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
>>> from io import StringIO
>>> print(type(StringIO().getvalue()))
<type 'unicode'>
>>>
>>> sys.stdout = out = StringIO()
>>> print("test")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unicode argument expected, got 'str'

```

`six.StringIO` aliases compatible libraries for each version so I'll use that to avoid the try.